src/test/scala/chiselTests/RecordSpec.scala

```diff
@@ -146,6 +146,34 @@ class RecordSpec extends ChiselFlatSpec with RecordSpecUtils with Utils {
     e.getMessage should include("contains aliased fields named (bar,foo)")
   }
 
+  they should "be OpaqueType for single element maps" in {
+    class SingleElementRecord extends Record {
+      private val underlying = UInt(8.W)
+      val elements = SeqMap("" -> underlying)
+      override def opaqueType = elements.size == 1
+      override def cloneType: this.type = this
+
+      def +(that: SingleElementRecord): SingleElementRecord = {
+        val _w = Wire(new SingleElementRecord)
+        _w.underlying := this.underlying + that.underlying
+        _w
+      }
+    }
+
+    val chirrtl = ChiselStage.emitChirrtl {
+      new Module {
+        val in1 = IO(Input(new SingleElementRecord))
+        val in2 = IO(Input(new SingleElementRecord))
+        val out = IO(Output(new SingleElementRecord))
+
+        out := in1 + in2
+      }
+    }
+    chirrtl should include("input in1 : UInt<8>")
+    chirrtl should include("input in2 : UInt<8>")
+    chirrtl should include("add(in1, in2)")
+  }
+
```

Comment:
Would `out := 0.U(8.W)` be expected to work?

Reply:
(And same question for an example where the RHS is not a Literal if it matters)

Reply:
I would say no cause the type of `out` is not actually being changed, only the IR being emitted after module close time will change

Reply:
Correct, this still enforces the Chisel type system.

Reply:
to clarify, I have been assuming this case would _not_ work and implementing my own use case accordingly, but, if I ever have to do something similar again, it would save me a ton of time if this did work anywhere `opaqueType` is true

Reply:
> it would save me a ton of time if my above example did work

This would sort of defeat the purpose of what an `OpaqueType` is supposed to be. If you want the type to act like the type it is wrapping, then why not just use a type alias? You could also alternatively use an implicit conversion.

Reply:
You can't use an implicit conversion because `final. def :=(that: Data)` accepts all data

Reply:
> then why not just use a type alias?

Because I have specific, runtime determined configuration where the number of elements is 1 (and `opaqueType` would evaluate to `true`). Only in those exact cases do i want `happensToBeOpaqueHere := UInt(8.W)` to work, in all other instances a firrtl binding error is appropriate

Reply:
And again, do not consider this a blocking request, i have already worked around it by manually calling another method instead of `:=` at all call sites. But that could have been avoided.

I further bring it up because the PR message says "and substitute type of the subfield for type of the Record" and I would like to understand what that means in practice

Reply:
> "only the IR being emitted after module close time will change"

is helpful in narrowing the scope of the answer but I do not really understand what that means to user code. in what situations does FIRRTL make a different decision about allowable bindings based on that change? or none?

Reply:
So the way this is implemented in this PR is by handling `OpaqueType` as a special case of `Record` - there's a custom type under the hood that does nothing other than note that a `Record` instance needs to be represented differently in the IR. This means that `OpaqueType` is not fully a native type (yet), and I think while it's _possible_ to support the monoconnect use case, that would mean adding special handling for that in connects which would currently be very hacky IMO

Reply:
Also I think the monoconnect case might fall naturally once we have more ingrained support for `OpaqueType`

Reply:
> which would currently be very hacky IMO
 thanks for the explanation, sounds best to wait for it